stdlib.ispc

```diff
@@ -5450,3 +5450,66 @@ __declspec(safe) static inline void assume(uniform bool test) {
     __do_assume_uniform(test);
     return;
 }
+
+///////////////////////////////////////////////////////////////////////////
+// Dot product and accumulate
+// Multiply groups of 4 adjacent pairs of unsigned 8-bit integers in a with corresponding signed
+// 8-bit integers in b, producing 4 intermediate signed 16-bit results.
+// Sum these 4 results with the corresponding 32-bit integer in acc, and return the result.
+__declspec(safe) static inline varying int32 dot4add_u8i8packed(varying int32 a, varying int32 b, varying int32 acc) {
+    if (__have_dot_product_vnni) {
+        return __dot4add_u8i8packed(a, b, acc);
+    } else {
+        int16 tmp1 = (int16)((a >> 24) & 0xFF) * (int16)((int8)(b >> 24));
+        int16 tmp2 = (int16)((a >> 16) & 0xFF) * (int16)((int8)(b >> 16));
+        int16 tmp3 = (int16)((a >> 8) & 0xFF) * (int16)((int8)(b >> 8));
+        int16 tmp4 = (int16)(a & 0xFF) * (int16)((int8)b);
+        return (int32)tmp1 + (int32)tmp2 + (int32)tmp3 + (int32)tmp4 + acc;
```

Comment:
I concerned that this code produces same result as HW instruction. I tried to print in console result for both branches with targets `avx2-i32x8` and `sse4-i32x8` and got the different results for `./tests/dot4add_u8i8-1.ispc` test.

Reply:
You probably saw the different results starting from the middle. I made last-minute changes to the tests and overlooked updating the iteration space, which was set to twice its intended size. Please give it another try now.

Reply:
Yes, they are same now.